### DISCUSSION TITLE
idea to improve searching

### DISCUSSION THREAD

**Discussion by orlfman** [stevenbenner/pacfinder#4] [Ideas]
Right now search appears to only look for filenames. I'm wondering if its possible to also include item descriptions into consideration for search results as well. Maybe have it where exact filename is position at the top with another category underneath those results that displays others with matching results in their descriptions.

**Comment by stevenbenner** [comment]
> I'm wondering if its possible to also include item descriptions into consideration for search results as well.

Yes, easily.

> Maybe have it where exact filename is position at the top with another category underneath those results that displays others with matching results in their descriptions.

That is a bit more challenging. As it would require creating a dedicated search screen or a custom list control.

I'm a bit unsure of where to take the search feature from here. A couple people had expressed interest in the ability to query the AUR, which would add a lot more complexity and probably necessitate building a more complete search tool rather than the single input field that exists today. However, I think it may be best to keep the search mechanism simple so that it remains fast and the number of results for partial queries (e.g. "pac") don't grow too large.

Some software has an "advanced search" feature in addition to a basic search. Maybe that would make sense here?

For now, yeah I think querying descriptions for a search match could be a good improvement. Though maybe it should be a configurable option?
